A72-open-telemetry-tracing.md

```diff
@@ -105,94 +133,181 @@ class OpenTelemetryPluginBuilder {
+  * key `message.event.type` with string value "SENT".
+  * key `message.message.id` with integer value of the seq no. The seq no. is a sequence
+    of integer numbers starting from 0 to identify sent messages within the stream, the same below.
+  * key `message.event.size.uncompressed` with integer value of uncompressed 
+    message size. The size is the total attempt message bytes without encryption,
+    not including grpc or transport framing bytes, the same below.
+  * If any compression, key `message.event.size.compressed` with integer value
+    of compressed message size.
+* When an inbound message has been received from the transport, add an Event
+  with name "Inbound message read" and the following attributes:
+  * key `message.event.type` with String value "RECEIVED".
+  * key `message.message.id` with integer value of the seq no.
+  * key `message.event.size.compressed` with integer value of wire message size. 
+  * If any compression, key `message.event.size.uncompressed` with integer value
+    of uncompressed message size.
 * When the stream is closed, set RPC status and end the attempt span.
 
 At the server:
-* When an outbound message has been sent, add message events to capture seq no.,
-  type(SENT) and uncompressed message size, and compressed message size if any compression.
-* When an inbound message has been read from the transport, add message events
-  to capture seq no., type(Received), wire message size, and uncompressed message size,
-  if any decompression.
+* When the application sends an outbound message to the transport, add an Event
+  with name "Outbound message sent" and the following attributes:
+  * key `message.event.type` with string value "SENT".
+  * key `message.message.id` with integer value of the seq no.
+  * key `message.event.size.uncompressed` with integer value of uncompressed
+    message size.
+  * If any compression, key `message.event.size.compressed` with integer value
+    of compressed message size.
+* When an inbound message has been read from the transport, add an Event
+  with name "Inbound message read" and the following attributes:
+  * key `message.event.type` with string value "RECEIVED".
+  * key `message.message.id` with integer value of the seq no.
+  * key `message.event.size.compressed` with integer value of wire message size.
+  * If any compression, key `message.event.size.uncompressed` with integer value
+    of uncompressed message size.
 * When the stream is closed, set the RPC status and end the span.
 
-Note that C++ is missing the seq no. information. And Java has issue of reporting
-decompressed message size upon receiving messages,
+### Limitations
+Note that C++ is missing the seq no. information due to lack of transport support.
+While it's not critical, we can include these information if users request it in the future.
+
+Java has an issue of reporting decompressed message size upon receiving messages,
```

Comment:
Maybe we should be slightly less specific here?  Rather than saying that Java will do this differently, maybe we should say that implementations that cannot report both the uncompressed size at the transport layer will instead add this second event.

The reason I suggest this is that I think we may have a similar concern in C-core, where we don't uncompress until above the transport.  @yashykt

Reply:
I think we should remove the mention of "transport" and we should use a more loose language similar to what we added in https://github.com/grpc/proposal/blob/master/A66-otel-stats.md. 

For example, in gRPC Core, the message events (both compressed and decompressed) are added in the message compression filter which is above the transport.

One thing I want to stress and want mentioned in the gRPC is that I want the flexibility to report the compressed and decompressed events separately. This will allow users to see the amount of time it took to do compression/decompression. For implementations that cannot do this, it's fine to use a single event or not report at all.

Reply:
**Send/receive messages to/from where**
IIRC In Java it is different than C++: reporting compressed/uncompressed happens _at_ the transport framer/deframer, except for reporting inbound uncompressed message sizes that happens later above transport. 
Earlier Mark had question about the context of send/receive, like it is pass flow control or on the wire. So I mentioned application. Since C++ does that above transport, I will not mention "to transport".

**Combined or separate events for reporting compressed and uncompressed message size**
C++ is already doing that separately. Java needs some changes to support reporting separate messages on before-compression at sending. Similar for Go I believe.